src/analysis/SmartDetectorsAnalysis/AutomaticResolutionState.cs

```diff
@@ -0,0 +1,27 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AutomaticResolutionState.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Azure.Monitoring.SmartDetectors.MonitoringAppliance.Analysis
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// A class encapsulating the automatic resolution state, used to extract all relevant
+    /// alert information for supplying the detector on automatic resolution checks.
+    /// </summary>
+    public class AutomaticResolutionState
+    {
+        /// <summary>
+        /// Gets or sets the alert ID this state relates to.
+        /// </summary>
+        public string AlertId { get; set; }
```

Comment:
>AlertId [](start = 22, length = 7)

Consider removing this since this is the state key

Reply:
done

---
In reply to: [235360743](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/27#discussion_r235360743) [](ancestors = 235360743)